code/modules/projectiles/guns/energy.dm

```diff
@@ -43,10 +43,30 @@
 	if(.)
 		update_icon()
 
-/obj/item/gun/energy/emp_act(severity)
+/obj/item/gun/energy/emp_act(var/severity)
 	..()
+	disable_cell_temp(severity)
 	queue_icon_update()
 
+/obj/item/gun/energy/proc/disable_cell_temp(var/severity)
+	set waitfor = FALSE
+	if(!power_supply)
+		return
+	var/mob/M
+	if(ismob(loc))
+		M = loc
+	var/initial_charge = power_supply.charge
+	power_supply.charge = 0
+	if(M && loc == M)
```

Comment:
Second condition is always TRUE.

Just move 61-62 under the ismob()

Reply:
It is not always true. The gun can be dropped while the sleep is running.